### PR TITLE
UI for mobile to mobile local pairing

### LINKED ISSUE #13859: UI for mobile to mobile local pairing
# Feature Issue
This issue is made to keep a track of the progress on building UI for local pairing mobile - mobile case.

Design links are here : https://www.figma.com/file/aztkse2iYuMAjVJRAsxqFO/Syncing-Settings?node-id=0%3A1&t=iOo0P7DpqBPbLbyf-0

@Samyoul

### PR DESCRIPTION
**Pull request by siddarthkay**
partially fixes #13859

### Summary
- A minimalistic UI to test out local pairing 
- Native modules in both ios and android bridge to call Go logic

Documentation of workflow : https://notes.status.im/4VbZ0PlKR0GBE4d_A-qG_A

Design is here : https://www.figma.com/file/aztkse2iYuMAjVJRAsxqFO/Syncing-Settings?node-id=0%3A1&t=iOo0P7DpqBPbLbyf-0

### Review notes
- This feature does not have a perfect UX at the moment.
- This introduces the local pairing two devices to pair.
- After successful pairing the receiver device can not log into the paired account yet (currently some things are being ironed out at status-go end)

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
none, this feature is under a toggle and all files are separate.

##### Functional
- account recovery

You will need mobile devices to test this feature
1) Sender : The device which contains the account data we want to pair
2) Receiver : The device which will be paired with the sender account and may or may-not be a new device

### Steps to test
- Open Status on Sender Device 
- Log in to your account
- Go to profile settings > Advanced > turn the toggle on for local pairing(alpha)
- Navigate back and scroll down to settings page and tap on Syncing menu item
- Tap on Sync new device
- Tap on Setup Syncing
- Enter the password of your account
- You will now see a QR code 

- Open Status on Receiver Device
- Log in to your account
- Go to profile settings > Advanced > turn the toggle on for local pairing(alpha)
- Navigate back and scroll down to settings page and tap on Sign out
- Tap on 3 dots on top right on login screen
- Tap on scan sync code
- Scan the QR code which you see on Receiver Device

- Wait for a few seconds and the devices should be paired.
- Close the app and when you re-open the app, under your keys you should see new account show up over there.
- Do not try to log-in ( there is an issue with paired database which is being addressed on status-go side).

status: ready

### CODE REVIEW

File: src/status_im/ui/screens/profile/user/views.cljs [hunk 1]
```diff
@@ -167,6 +168,12 @@
        :accessibility-label :about-button
        :chevron             true
        :on-press            #(re-frame/dispatch [:navigate-to :about-app])}]
+     (when local-pairing-mode-enabled? [quo/list-item
+                                        {:icon                :main-icons/info
+                                         :title               (i18n/label :t/syncing)
+                                         :accessibility-label :synching
```
Comment (ibrkhalil) [inline src/status_im/ui/screens/profile/user/views.cljs:174]: `:syncing`?

**Comment by siddarthkay** [comment]
Attempting to rebase and get this merged in its current state @cammellos , 
We will address further work / issues in new PRs :)

**Comment by flexsurfer** [comment]
Files changed 1,355 O_o ?

**Comment by siddarthkay** [comment]
> Files changed 1,355 O_o ?

yeah I messed up and realised it too late.

I am making a new PR.

Will close this one.

**Comment by siddarthkay** [comment]
closing in favour of  https://github.com/status-im/status-mobile/pull/14514

**Pull request closed without merging**